### PR TITLE
Adding set options functionality

### PR DESCRIPTION
**Pull request by FranciscoMolinaKM**
Added functions on Stellar.Base.Operation to allow the set options operation
Added functions on Stellar.Base.Signer to allow the parse into XDR of the signer
Modify the SetOptionsOp module on Stellar.XDR.Types.Transaction to allow the optional use of an inflation destination account
Added all the test to build the SetOptions transaction
Added all the test to parses the signer
Added the test to sign the setOptions operation

### CODE REVIEW

File: lib/stellar/base/operation.ex [hunk 1]
```diff
@@ -353,8 +393,7 @@ defmodule Stellar.Base.Operation do
   end
 
   def to_xdr(%{type: type} = this) when type == type_allow_trust() do
-    with {:ok, trustor} <-
-           KeyPair.from_public_key(this.trustor) |> KeyPair.to_xdr_accountid(),
+    with {:ok, trustor} <- KeyPair.from_public_key(this.trustor) |> KeyPair.to_xdr_accountid(),
```
[the remark this answers is not in the document]
Comment (FranciscoMolinaKM) [reply lib/stellar/base/operation.ex]: The mix format modifies this line in that way.

File: lib/stellar/base/operation.ex [hunk 2]
```diff
@@ -377,10 +416,8 @@ defmodule Stellar.Base.Operation do
              limit: this.limit
            }
            |> ChangeTrustOp.new(),
-         {:ok, change_trust_body} <-
-           OperationBody.new({:CHANGE_TRUST, change_trust_op}),
-         {:ok, operation} <-
-           %XDROperation{body: change_trust_body} |> XDROperation.new() do
+         {:ok, change_trust_body} <- OperationBody.new({:CHANGE_TRUST, change_trust_op}),
+         {:ok, operation} <- %XDROperation{body: change_trust_body} |> XDROperation.new() do
```
[the remark this answers is not in the document]
Comment (FranciscoMolinaKM) [reply lib/stellar/base/operation.ex]: The mix format modifies this line in that way.

File: lib/stellar/base/operation.ex [hunk 3]
```diff
@@ -434,12 +512,26 @@ defmodule Stellar.Base.Operation do
...
+  end
+
   def is_valid_amount(0, false), do: false
   def is_valid_amount(amount, _) when amount < 0, do: false
   def is_valid_amount(amount, _) when is_integer(amount), do: true
 
   def is_valid_amount(amount, _) when is_float(amount) do
-    ((amount * unit()) |> Kernel.trunc()) / unit() ==
-      amount
+    ((amount * unit()) |> Kernel.trunc()) / unit() == amount
```
[the remark this answers is not in the document]
Comment (FranciscoMolinaKM) [reply lib/stellar/base/operation.ex]: The mix format modifies this line in that way.

File: lib/stellar/base/transaction_builder.ex [hunk 4]
```diff
@@ -76,8 +76,7 @@ defmodule Stellar.Base.TransactionBuilder do
 
   def build(this) do
     with updated_source_account <- Account.increment_sequence_number(this.source_account),
-         {:ok, xdr_transaction} <-
-           build_transaction_xdr(this, updated_source_account),
+         {:ok, xdr_transaction} <- build_transaction_xdr(this, updated_source_account),
```
[the remark this answers is not in the document]
Comment (FranciscoMolinaKM) [reply lib/stellar/base/transaction_builder.ex]: The mix format modifies this lines in that way.

File: test/stellar/base/account_test.exs [hunk 5]
```diff
@@ -24,8 +24,7 @@ defmodule Stellar.Base.Account.Test do
       assert Stellar.Base.Account.accountId(account) ==
                "GBBM6BKZPEHWYO3E3YKREDPQXMS4VK35YLNU7NFBRI26RAN7GI5POFBB"
 
-      assert Stellar.Base.Account.sequence_number(account) ==
-               100
+      assert Stellar.Base.Account.sequence_number(account) == 100
```
[the remark this answers is not in the document]
Comment (FranciscoMolinaKM) [reply test/stellar/base/account_test.exs]: The mix format modifies these lines in that way.

File: lib/stellar/base/operation.ex [hunk 6]
```diff
@@ -422,6 +459,47 @@ defmodule Stellar.Base.Operation do
     end
   end
 
+  @doc """
+  Parses the map with the info to XDR, and define the data as SetOptionsOp struct in this case not all the info
+   must be converted to XDR, only the inflation destination and the signer the other values can pass as defalut type
```
[the remark this answers is not in the document]
Comment (andreaCuesta) [reply lib/stellar/base/operation.ex]: @FranciscoMolinaKM, please fix the word `defalut` is default

File: test/stellar/base/signer_test.exs [hunk 7]
```diff
@@ -68,4 +68,46 @@ defmodule Stellar.Signer.Test do
...
+              key: "GDDVWKPMJKUH766SMOVKLDTZQCC4B7Q42YRRH7YBBDYDFPI7LWKJP55F",
+              weight: weight
+            })
+          )
+        end)
+
+      assert result == true
+    end
+
+    test "parse a signer with a invalid weigth" do
```
[the remark this answers is not in the document]
Comment (andreaCuesta) [reply test/stellar/base/signer_test.exs]: @FranciscoMolinaKM, please fix the word `weigth` the correct one is `weight`

File: test/stellar/base/transaction_builder_test.exs [hunk 8]
```diff
@@ -269,4 +269,176 @@ defmodule Stellar.Base.TransactionBuilder.Test do
...
+            high_threshold: 1
+          })
+        )
+        |> TransactionBuilder.set_timeout(10)
+        |> TransactionBuilder.build()
+
+      assert status == :ok
+      assert List.first(transaction.operations).homeDomain == "kommit.co"
+      assert updated_account._accountId == source._accountId
+    end
```
Comment (andreaCuesta) [inline test/stellar/base/transaction_builder_test.exs]: @FranciscoMolinaKM, please add an assert to check the `high_threshold`

**Pull request merged**